go/libraries/doltcore/env/actions/infer_schema.go

```diff
@@ -181,7 +182,19 @@ func leastPermissiveType(strVal string, floatThreshold float64) typeinfo.TypeInf
 		return typeinfo.BoolType
 	}
 
-	return typeinfo.StringDefaultType
+	if strings.Contains(strVal, "{") {
+		var json map[string]interface{}
+		err := json2.Unmarshal([]byte(strVal), &json)
```

Comment:
`json2` could benefit from a better name. I'd recommend not renaming the package import, so you can keep that as `json`, and just use a different variable name for `json` on line 186. It looks like that single variable is the only conflict with the `json` package name, so should be easy to change. 

Alternatively though, if you do need to rename the package, something like `jsonenc` or `encjson` would be a little more readable than `json2`.